src/services/sync/strategies/TSP/TSPSchoolSyncer.js

```diff
@@ -138,36 +158,90 @@ class TSPSchoolSyncer extends mix(Syncer).with(ClassImporter) {
 				`School has ${schoolTeachers.length} teachers, ${schoolStudents.length} students` +
 					`, and ${schoolClasses.length} classes.`
 			);
+			const teacherMapping = await this.processTSPTeachers(schoolTeachers, school);
+			const classMapping = await this.processTSPStudents(schoolStudents, school);
+			this.logInfo('Syncing classes...');
+			// create classes based on API response and user mappings
+			await this.createOrUpdateClasses(schoolClasses, school, teacherMapping, classMapping);
 
-			const teacherMapping = {};
-			const classMapping = {};
+			this.logInfo('Done.');
+		} catch (err) {
+			this.logError('Error while syncing process TSP Schools', { error: err });
```

Comment:
We have multiple times this 2 lines. Can you please outsource it i a private method like handleError('string', err)

Reply:
Done